# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1238586555 -7200
#      Wed Apr 01 13:49:15 2009 +0200
# Node ID 46a0e6e9e4465508d0e248fd88beb71b5f63efd3
# Parent  17ce09c3d0304c5b91390b74f13188a0e83759b7
fix invalid subscript in symtab.cc

diff --git a/src/symtab.cc b/src/symtab.cc
--- a/src/symtab.cc
+++ b/src/symtab.cc
@@ -181,17 +181,17 @@ out_of_date_check_internal (octave_funct
 		  int nm_len = nm.length ();
 
 		  std::string file;
 		  std::string dir_name;
 
 		  if (octave_env::absolute_pathname (nm)
 		      && ((nm_len > 4 && (nm.substr (nm_len-4) == ".oct"
 					  || nm.substr (nm_len-4) == ".mex"))
-			  || (nm_len > 2 && nm.substr (nm_len-4) == ".m")))
+			  || (nm_len > 2 && nm.substr (nm_len-2) == ".m")))
 		    file = nm;
 		  else
 		    {
 		      // We don't want to make this an absolute name,
 		      // because load_fcn_file looks at the name to
 		      // decide whether it came from a relative lookup.
 
 		      if (! dispatch_type.empty ())
